client/my-sites/site-settings/test/form-general.jsx

```diff
@@ -88,4 +131,84 @@ describe( 'SiteSettingsFormGeneral ', () => {
 			expect( comp.find( 'Banner' ).length ).toBe( 0 );
 		} );
 	} );
+
+	describe( 'Privacy picker - simple sites', () => {
+		let testProps;
+
+		beforeEach( () => {
+			testProps = {
+				...props,
+				siteIsJetpack: false,
+				site: { plan: PLAN_PERSONAL },
+				fields: {
+					blog_public: 1,
+					wpcom_coming_soon: 0,
+				},
+				withComingSoonOption: true,
+				updateFields: fields => {
+					testProps.fields = fields;
+				},
+			};
+		} );
+
+		test( `Should have 4 visibility options`, () => {
+			const { container } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+			expect( container.querySelectorAll( '[name="blog_public"]' ).length ).toBe( 4 );
+		} );
+
+		[
+			[ 'Coming soon', '.is-coming-soon [name="blog_public"]', 1 ],
+			[ 'Public', '.is-public [name="blog_public"]', -1 ],
+			[ 'Hidden', '.is-hidden [name="blog_public"]', -1 ],
+			[ 'Private', '.is-private [name="blog_public"]', 1 ],
+		].forEach( ( [ name, selector, initialBlogPublic ] ) => {
+			test( `${ name } option should be selectable`, () => {
+				testProps.fields.blog_public = initialBlogPublic;
+				const { container, rerender } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+
+				const radioButton = container.querySelector( selector );
+				expect( radioButton.checked ).toBe( false );
+				fireEvent.click( radioButton );
+				rerender( <SiteSettingsFormGeneral { ...testProps } /> );
+				expect( radioButton.checked ).toBe( true );
```

Comment:
I think this rerender and assertion is really just testing the logic of the fake `updateFields` function in `testProps`? Since `<SiteSettingsFormGeneral>` is a fully controlled component it might be better to skip the rerender and use a `jest.fn()` to assert the `updateFields` prop is called with the correct parameters?